packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebViewFactory.java

```diff
@@ -5,29 +5,24 @@
 package io.flutter.plugins.webviewflutter;
 
 import android.content.Context;
-import android.view.View;
-import io.flutter.plugin.common.BinaryMessenger;
-import io.flutter.plugin.common.MethodChannel;
 import io.flutter.plugin.common.StandardMessageCodec;
 import io.flutter.plugin.platform.PlatformView;
 import io.flutter.plugin.platform.PlatformViewFactory;
-import java.util.Map;
 
-public final class FlutterWebViewFactory extends PlatformViewFactory {
-  private final BinaryMessenger messenger;
-  private final View containerView;
+class FlutterWebViewFactory extends PlatformViewFactory {
+  private final InstanceManager instanceManager;
 
-  FlutterWebViewFactory(BinaryMessenger messenger, View containerView) {
+  FlutterWebViewFactory(InstanceManager instanceManager) {
     super(StandardMessageCodec.INSTANCE);
-    this.messenger = messenger;
-    this.containerView = containerView;
+    this.instanceManager = instanceManager;
   }
 
-  @SuppressWarnings("unchecked")
   @Override
   public PlatformView create(Context context, int id, Object args) {
-    Map<String, Object> params = (Map<String, Object>) args;
-    MethodChannel methodChannel = new MethodChannel(messenger, "plugins.flutter.io/webview_" + id);
-    return new FlutterWebView(context, methodChannel, params, containerView);
+    final PlatformView view = (PlatformView) instanceManager.getInstance((Integer) args);
+    if (view == null) {
+      throw new IllegalStateException("Unable to find WebView instance: " + args);
+    }
+    return view;
   }
```

Comment:
The main reason I didn't want create to the WebView in the `PlatformViewFactory` was that I wanted to eventually have an update for https://github.com/flutter/flutter/issues/51875. Where a user would be able to load a URL for a `WebView` and make calls to scrolling without needing to add the `Widget` to the widget tree. We probably should actually have the `PlatformView` API provide support to create the View without adding it to the tree, but I think this is the only way around it at the moment.

Also, the Pigeon API is built to mimic the [android.webkit](https://developer.android.com/reference/android/webkit/WebView) API. My goal was to avoid significant deviation from this because it can make it harder to handle changes to the API later or run into conflicts when adding features we weren't aware of before.

In the long term, it should become a standard to always call `create` in the constructor of a class and `dispose` would be handled automatically by garbage collection. This is probably at least 2 quarters away, but the changes to `webview_flutter` are essentially the stepping stones to get there.